src/components/organisms/PaymentModal/PaymentForm.tsx

```diff
@@ -61,32 +64,69 @@ const PaymentForm: React.FunctionComponent<PropsType> = ({
     }
   }, [event.id, venueId, ticketPrice, user, selectedTab]);
 
-  const handleSubmit = async (e: React.FormEvent<HTMLFormElement>) => {
+  const handleSubmit = async (
+    e: React.MouseEvent<HTMLButtonElement, MouseEvent>,
+    setIsLoading: (isLoading: boolean) => void,
+    operations: (stripe: Stripe, elements: StripeElements) => Promise<void>
+  ) => {
     if (!user) return;
-    setIsFormBeingSubmitted(true);
+    setIsLoading(true);
     e.preventDefault();
     if (!stripe || !elements) {
       setErrorMessage("Oops something wrong happened");
+      setIsLoading(false);
       return;
     }
-    const result = await stripe.confirmCardPayment(clientSecret || "", {
+    await operations(stripe, elements);
+    setIsLoading(false);
+  };
+
+  const handlePayment = (e: React.MouseEvent<HTMLButtonElement, MouseEvent>) =>
+    handleSubmit(e, setIsPaymentProceeding, pay);
+  const handleSaveCard = (e: React.MouseEvent<HTMLButtonElement, MouseEvent>) =>
+    handleSubmit(e, setIsCardBeingSaved, saveCardAndPay);
+
+  const pay = async (stripe: Stripe, elements: StripeElements) => {
+    if (!(user?.email && clientSecret)) {
+      setErrorMessage("Oops something wrong happened");
+      return;
+    }
+    const result = await stripe.confirmCardPayment(clientSecret, {
       payment_method: {
         card: elements.getElement(CardElement) || { token: "" },
         billing_details: {
-          email: user.email ?? undefined,
+          email: user.email,
         },
       },
-      receipt_email: user.email ?? undefined,
+      receipt_email: user.email,
     });
 
     if (result.error) {
       setErrorMessage(result.error.message);
-    } else {
-      if (result.paymentIntent?.status === "succeeded") {
-        setIsPaymentSuccess(true);
+    } else if (result.paymentIntent?.status === "succeeded") {
```

Comment:
if the status is not "succeeded" we should probably inform the user via an error message.